v1-v4: LoadMany() returns in non-deterministic order

In `v1-v4` of this library, `LoadMany()` does not return in deterministic order. That is, a call to `LoadMany("A", "B")` can give results `[ResultA, ResultB]`.

To reproduce, run the unit tests multiple times:
```
$ go test -count=1000 .
--- FAIL: TestLoader (0.00s)
    --- FAIL: TestLoader/test_LoadMany_returns_len(errors)_==_len(keys) (0.02s)
        dataloader_test.go:103: Expected an error on the first item loaded
        dataloader_test.go:107: Expected second and third errors to be nil
--- FAIL: TestLoader (0.00s)
    --- FAIL: TestLoader/test_LoadMany_returns_len(errors)_==_len(keys) (0.02s)
        dataloader_test.go:103: Expected an error on the first item loaded
        dataloader_test.go:107: Expected second and third errors to be nil
--- FAIL: TestLoader (0.00s)
    --- FAIL: TestLoader/test_LoadMany_returns_len(errors)_==_len(keys) (0.02s)
        dataloader_test.go:103: Expected an error on the first item loaded
        dataloader_test.go:107: Expected second and third errors to be nil
--- FAIL: TestLoader (0.00s)
    --- FAIL: TestLoader/test_LoadMany_returns_len(errors)_==_len(keys) (0.02s)
        dataloader_test.go:103: Expected an error on the first item loaded
        dataloader_test.go:107: Expected second and third errors to be nil
--- FAIL: TestLoader (0.00s)
    --- FAIL: TestLoader/test_LoadMany_returns_len(errors)_==_len(keys) (0.02s)
        dataloader_test.go:103: Expected an error on the first item loaded
        dataloader_test.go:107: Expected second and third errors to be nil
--- FAIL: TestLoader (0.00s)
    --- FAIL: TestLoader/test_LoadMany_returns_len(errors)_==_len(keys) (0.02s)
        dataloader_test.go:103: Expected an error on the first item loaded
        dataloader_test.go:107: Expected second and third errors to be nil
```

It seems like the cause may be the `thunk` creation happens concurrently in a goroutine, so the batch of keys passed to the batch function may end up out of order.

--- a/dataloader.go
+++ b/dataloader.go
@@ -241,14 +241,16 @@ func (l *Loader) LoadMany(keys []string) ThunkMany {
 	wg := sync.WaitGroup{}
 
 	wg.Add(length)
-	for i := range keys {
-		go func(i int) {
+	for i, key := range keys {
+		// Request the key serially to ensure the result order is maintained in key order.
+		thunk := l.Load(key)
+
+		go func(i int, thunk Thunk) {
 			defer wg.Done()
-			thunk := l.Load(keys[i])
 			result, err := thunk()
 			data[i] = result
 			errors[i] = err
-		}(i)
+		}(i, thunk)
 	}
 
 	go func() {

